### ISSUE TITLE
What are the key parameters ?

### ISSUE THREAD

**Issue by Cain399** [ml-jku/mhn-react#1]
Hello, your work is of great help to my research. Now I have a new data set and want to use this model, but the effect is not very good. May I ask which parameters play a key role and influence on the model performance?  Looking forward to your reply .

**Comment by phseidl** [comment]
Hi @Cain399,
thanks for your interest in our work.
Could you be more specific about the new data set and your intended use? Are you using it as zero-shot model, fine-tuning or training a model from scratch? Also are you particularly interested in top-1 accuracy or other metrics?

**Comment by Cain399** [comment]
Thank you for your reply. What I use is a data set similar to USTPO, in which specific responses are different in the same form.  I trained a new model from scratch, hoping that the top-1 index would be better, and now it's only around 10%.

**Comment by phseidl** [comment]
Can you share the dataset, so I could investigate what is going on? Have you looked at training behavior? Hyperparameters might need to be adjusted.

**Comment by Cain399** [comment]
Sorry, this data set is provided by the company and cannot be disclosed for the time being.  This data set is characterized by a complex structure of compounds with a number of around 100,000.  My data format is processed to the same format as USTPO_50K_mHN_prepro.csv.gz.  However, the template I extracted may be different from this one. I directly extracted the template using the extract_from_reaction method in rdchiral, and your method seems to select the template.  In the course of training, training loss has been declining and cannot converge, while verification loss will converge soon.  I would appreciate it if you could let me know which hyperparameters you can tweak to make a difference.

**Comment by phseidl** [comment]
Important hyperparameters are the beta scaling factor of the Hopfield layer β (e.g. 1/sqrt(d), the dimension of the association-space d (e.g. 1024). The association space should be normalized via layer-norm. (--norm_asso=True).
You can also compare it to a standard FF-NN by setting --model_type=segler

**Comment by Cain399** [comment]
Thank you very much. I will have try.

**Comment by phseidl** [comment]
Please send me an email if you need further assistance (seidl@ml.jku.at), or just to give me an update on your results.
